src/libexpr-c/nix_api_expr.cc

```diff
@@ -106,7 +107,15 @@ EvalState * nix_state_create(nix_c_context * context, const char ** lookupPath_c
             for (size_t i = 0; lookupPath_c[i] != nullptr; i++)
                 lookupPath.push_back(lookupPath_c[i]);
 
-        return new EvalState{nix::EvalState(nix::LookupPath::parse(lookupPath), store->ptr)};
+        // FIXME leak
+        auto * evalSettings = new nix::EvalSettings{nix::settings.readOnlyMode};
+
+        return new EvalState{
+            nix::EvalState(
+                nix::LookupPath::parse(lookupPath),
+                store->ptr,
+                *evalSettings)
+        };
```

Comment:
@roberth I don't think this is very good, but I am not sure what we should do here instead?

I do like the forward compat of the C bindings not exposing any settings or reading the global ones.